src/tools/rpmemd/rpmemd_log.c

```diff
@@ -34,7 +34,12 @@
  * rpmemd_log.c -- rpmemd logging functions definitions
  */
 /* for GNU version of basename */
+/* XXX Consider changing to Posix basename for consistency */
+#ifdef __FreeBSD__
+#include <libgen.h>
+#else
 #define _GNU_SOURCE
```

Comment:
This GNU_SOURCE macro can be a problem some in other platforms as well, I remember clang warning about unused macros...
I would suggest detecting GNUlibc in a Makefile, and define the _GNU_SOURCE macro in CFLAGS if needed.

Just a suggestion, it doesn't need to be in this PR yet.